TrackerGeom/inc/Tracker.hh

```diff
@@ -22,112 +21,66 @@
-    double z0()   const { return _z0;} // in Mu2e coordinates (everything else is in Tracker coordiates).
-    double zHalfLength() const;
+    // origin in nominal tracker coordinate system
+    const xyzVec& origin() const { return _origin; }
+
+    // common straw attributes that don't depend on the specific element
+    const StrawProperties& strawProperties() const { return _strawprops; }
 
-    // Tracker Element Accessors
+    // constituent access
     size_t nPlanes() const { return _planes.size(); }
     size_t nPanels() const { return _panels.size(); }
     size_t nStraws() const { return _straws.size(); }
-
-    // origin: defines nominal tracker coordinate system
-    const xyzVec& origin() const { return _origin; }
-
-    // constituent access
-    const Plane& plane( const StrawId& id ) const{
-      return _planes.at(id.getPlane());
-    }
-
-    PlaneCollection const& planes() const {
-      return _planes;
-    }
-    PanelCollection const& panels() const{
-      return _panels;
-    }
-    StrawCollection const& straws() const{
-      return _straws;
-    }
-
-    const Panel& panel( const StrawId& id ) const{
-      return _panels.at(id.uniquePanel());
-    }
-
-    uint16_t strawIndex(const StrawId& id) const {
-      return _strawindex.at(id.asUint16());
-    }
-
-    const Straw& straw( const StrawId& id) const{
-      return _straws[strawIndex(id)];
-    }
-
-// deprecated interface 
-    const Plane& getPlane( const StrawId& id ) const{
-      return _planes.at(id.getPlane());
-    }
-    
-    const Plane& getPlane( uint16_t n ) const{
-      return _planes.at(n);
-    }
-
-    PlaneCollection const& getPlanes() const {
-      return _planes;
-    }
-
-    const Panel& getPanel( const StrawId& id ) const{
-      return _panels.at(id.uniquePanel());
-    }
-
-    const Straw& getStraw( const StrawId& id) const{
-      return _straws[strawIndex(id)];
-    }
-
-    Straw& getStraw( const StrawId& id) {
-      return _straws[strawIndex(id)];
-    }
-    
-    StrawCollection const& getStraws() const{
-      return _straws;
-    }
-
-
-    // common attributes that don't depend on the specific element
-    const StrawProperties& strawProperties() const { return _strawprops; }
+    PlaneCollection const& planes() const { return _planes; }
+    PanelCollection const& panels() const{ return _panels; }
+    StrawCollection const& straws() const{ return _straws; }
+    // fast indexed lookup by StrawId through indirect map
+    uint16_t strawIndex(const StrawId& id) const { return _strawindex.at(id.asUint16()); }
+    const Plane& plane( const StrawId& id ) const{ return _planes.at(id.getPlane()); }
+    const Panel& panel( const StrawId& id ) const{ return _panels.at(id.uniquePanel()); }
+    const Straw& straw( const StrawId& id) const{ return _straws[strawIndex(id)]; }
+
+    // access the G4Tracker
+    const std::shared_ptr<G4Tracker>& g4Tracker() const { return _g4tracker; }
```

Comment:
I *think* that this actually lets the end user modify the underlying G4Tracker object since the const-ness applies to the stared_ptr<T> not to the pointee.    I *think* that the TrackerMaker relies on this on order to compile (friending does not get rid of const-ness ).

See my other comment the line where the member data _g4Tracker is declared.

I recommend the pair of functions:

   G4Tracker const* g4Tracker() const { return _g4tracker.get(); }
   G4Tracker*  g4Tracker() { return _g4tracker.get(); }

I think that all downstream code will continue to work without changes.  

My personal style would have been to return by reference not by value but that would require extensive changes to downstream code - so forget it.